fix(curriculum): add lesson to registration form practice project for adding method attribute (#48208)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/63541ef4f96cd82e8e6c788a.md

---
id: 63541ef4f96cd82e8e6c788a
title: Step 12
challengeType: 0
dashedName: step-12
---

# --description--

The `method` attribute specifies how to send form-data to page specified in the `action` attribute. The form-data can be sent as URL variables (with `method="get"`) or as HTTP post transaction (with `method="post"`).

Add `post` value to `method` attribute in `form`.

# --hints--

> 

Reviewer: ```suggestion

You should have one `form` with `method` attribute

```

Then add the following test 

\`\`\`js
assert.equal(document.querySelectorAll('form')?.getAttribute('method')?.length, 1);
\`\`\`

Reviewer: Maybe a third test, that checks for the amount of forms 🤔 

These test usually have a hint of the following.

> You shouldn't add a new `form` element.

Author: `assert.equal(document.querySelectorAll('form')?.getAttribute('method')?.length, 1);` -> return `Node` element 
Calling `getAttribute()` is not possible.. !! 
Should we iterate over each `form` element and validate ?
_Suggestion_:  Just validating for one `form` will be sufficient since validation for multiple form element will be done as first test !!

Reviewer: > Suggestion: Just validating for one form will be sufficient since validation if multiple form element are present or not will be done as first test !!

Sounds good go ahead 👍, I am happy as long as it isn't possible for user to add the code below

> ```html
> <form method='post'></form>
> <form action='https://register-demo.freecodecamp.org'></form>
> ```

Author: - [x] Added test case to check if multiple forms are added 
- [x] Added test case to check if method value is present or not

You should add `post` value to `method` attribute.

```js
assert.equal(document.querySelector('form')?.method, 'post');
```

# --seed--

## --seed-contents--

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8">
    <title>Registration Form</title>
    <link rel="stylesheet" href="styles.css" />
  </head>
--fcc-editable-region--
  <body>
    <h1>Registration Form</h1>
    <p>Please fill out this form with the required information</p>
    <form action='https://register-demo.freecodecamp.org'></form>
  </body>

> --fcc-editable-region--

Reviewer: ```suggestion
  <body>
    <h1>Registration Form</h1>
    <p>Please fill out this form with the required information</p>
--fcc-editable-region--
    <form action='https://register-demo.freecodecamp.org'></form>
--fcc-editable-region--
  </body>

Author: Agree on this.. Will fix this

Author: Done

---
id: 63541ef4f96cd82e8e6c788a
title: Step 12
challengeType: 0
dashedName: step-12
---

# --description--

> The `method` attribute specifies how to send form-data to page specified in the `action` attribute. The form-data can be sent as URL variables (with `method="get"`) or as HTTP post transaction (with `method="post"`).

Reviewer 2: ```suggestion
The `method` attribute specifies how to send form-data to page specified in the `action` attribute. The form-data can be sent via a `GET` request as URL parameters (with `method="get"`) or via a `POST` request as data in the request body (with `method="post"`).
```

I think we should clarify this a bit.

> The `method` attribute specifies how to send form-data to page specified in the `action` attribute. The form-data can be sent via a `GET` request as URL parameters (with `method="get"`) or via a `POST` request as data in the request body (with `method="post"`).

Reviewer 2: ```suggestion
The `method` attribute specifies how to send form-data to the URL specified in the `action` attribute. The form-data can be sent via a `GET` request as URL parameters (with `method="get"`) or via a `POST` request as data in the request body (with `method="post"`).
```
Grammar nit

> Add `post` value to `method` attribute in `form`.

Reviewer 2: ```suggestion
Set the `method` attribute to send your form data via a `POST` request.
```

# --hints--

You shouldn't add a new `form` element.

```js
assert.equal(document.querySelectorAll('form').length, 1 )
```

> You should have one `form` with `method` attribute

Reviewer 2: ```suggestion
Your `form` element should have a `method` attribute.
```

```js

> assert.notEqual(document.querySelector('form')?.getAttribute('method'), null);

Reviewer 2: ```suggestion
assert.exists(document.querySelector('form')?.getAttribute('method'));
```
Prevents both `null` and `undefined`.

```

> You should add `post` value to `method` attribute.

Reviewer 2: ```suggestion
Your `method` attribute should be set to `post`.
```
